build/cdc/dev/Dockerfile.website.cdc.dev

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Development Dockerfile for Custom Data Commons python webserver
+
+# Use official Python runtime as a parent image
+FROM python:3.12-slim
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Install dependencies and protoc
+RUN apt-get update && apt-get install -y wget unzip
+RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v23.3/protoc-23.3-linux-x86_64.zip -O protoc.zip \
+    && unzip protoc.zip -d /usr/local \
+    && rm protoc.zip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy python requirements file
+COPY server/requirements.txt /app/requirements.txt
+
+# Create the python virtual environment and install dependencies
+RUN python3 -m venv /app/.env \
+    && /app/.env/bin/pip install --upgrade pip \
+    && /app/.env/bin/pip install -r /app/requirements.txt
+
+# Copy source files to the container
+COPY server/. /app/server/
+COPY shared/. /app/shared/
+
+# Build subject page protos
+RUN protoc -I=/app/server/config/ --python_out=/app/server/config /app/server/config/subject_page.proto
```

Comment:
For prod containers, we've assumed that python proto stubs were already created. I guess here it makes sense to generate them in the container given it's for dev but perhaps think about whether it's necessary or not.

Reply:
I think for dev it's helpful to generate them in case we make any changes.